File: common/src/main/java/de/werkelmann/rekote/address/ip/IsIpRule.java
package de.werkelmann.rekote.address.ip;

import de.werkelmann.rekote.address.ConnectionRule;
import org.apache.commons.validator.routines.InetAddressValidator;

public class IsIpRule implements ConnectionRule {
    @Override
    public boolean isValid(String address) {
        return InetAddressValidator.getInstance().isValid(address);
    }
}

File: common/src/main/kotlin/de/werkelmann/rekote/address/ip/IsIpRule.kt
package de.werkelmann.rekote.address.ip

import de.werkelmann.rekote.address.ConnectionRule
import org.apache.commons.validator.routines.InetAddressValidator

class IsIpRule : ConnectionRule {
    override fun isValid(address: String): Boolean {
        return InetAddressValidator.getInstance().isValid(address)
    }
}
